operator-common/src/main/java/io/strimzi/operator/common/DefaultAdminClientProvider.java

```diff
@@ -23,34 +23,57 @@
     @Override
     public Admin createAdminClient(String hostname, Secret clusterCaCertSecret, Secret keyCertSecret, String keyCertName) {
         Admin ac;
-        PasswordGenerator pg = new PasswordGenerator(12);
-        String trustStorePassword = pg.generate();
-        File truststoreFile = Util.createFileTrustStore(getClass().getName(), "ts", Ca.cert(clusterCaCertSecret, Ca.CA_CRT), trustStorePassword.toCharArray());
+        String trustStorePassword = null;
+        File truststoreFile = null;
+        // provided Secret with cluster CA certificate for TLS encryption
+        if (clusterCaCertSecret != null) {
+            PasswordGenerator pg = new PasswordGenerator(12);
+            trustStorePassword = pg.generate();
+            truststoreFile = Util.createFileTrustStore(getClass().getName(), "ts", Ca.cert(clusterCaCertSecret, Ca.CA_CRT), trustStorePassword.toCharArray());
+        }
+
         try {
```

Comment:
I know this isn't under the remit of this PR, but I noticed that both of the `try`s here have no catch, I guess that means if for any reason they throw the error gets propagated up and the user would see this somehow? For instance in the case where `keyCertName + ".password"` isn't a valid key in the secret?

Reply:
I would see the reason about the error in the UO log. What's your proposal?

Reply:
I guess @samuel-hawker is wondering whether it makes sense to wrap the exception using a more meaningful exception type? My opinion on that (whether or not that is what he meant) was that it makes little difference. It's not going to be an exception which would be caught by the operator, so the only place it would appear would be in the logs and in the `status` as a reason for not being ready. I don't think anyone could reasonably do any automation around the problem in either of those places.

Reply:
If this is the case, then I agree that catching and re-throwing as a different exception makes a little difference in this case. @samuel-hawker any more comments?